### PR TITLE
Test initialization for nationwide states/pumas

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description
- *Category*: test
- *JIRA issue*: [MIC-3727](https://jira.ihme.washington.edu/browse/MIC-3727)
- *Research reference*: n/a

### Changes and notes
Since we have chosen to implement a single artifact for all population data,
the only thing needed to ensure states/pumas are being properly initialized
is to add a couple of integration tests.
* check that upon initialization, all states from the artifact exist in the state column
* check that upon initialization, each non-qg address_id is tied to a unique state/puma

### Verification and Testing
Tests are passing

### CODE REVIEW

File: integration_tests/conftest.py [hunk 1]
```diff
@@ -12,6 +12,9 @@ def sim() -> InteractiveContext:
     """Initialize a simulation for use in tests"""
     simulation = InteractiveContext(paths.MODEL_SPEC_DIR / "model_spec.yaml", setup=False)
     simulation.configuration.population.population_size = 20_000
+    simulation.configuration.input_data.artifact_path = str(
```
Comment (stevebachmeier) [inline integration_tests/conftest.py:15]: The entire test suite is passing with this change...so is it time to go ahead and start using this artifact as the "official" one? (keeping in mind it won't actually get to `main` for a bit while we work on the epic

**Review by hussain-jafari — APPROVED** [review]

File: integration_tests/conftest.py (hunk 1 above)
Comment (rmudambi) [reply integration_tests/conftest.py:15]: yes, that sounds good

File: integration_tests/test_household_structure.py [hunk 2]
```diff
@@ -39,3 +39,21 @@ def test_all_households_have_reference_person(tracked_live_populations):
...
+    assert states_in_artifact == set(initialized_pop["state"])
+
+
+def test_initialized_pumas_states(populations):
+    """Each unique non-GQ initialized address_id should have identical puma/state"""
+    initialized_pop = populations[0]
+    non_gq_pop = initialized_pop[
+        ~initialized_pop["household_id"].isin(data_values.GQ_HOUSING_TYPE_MAP)
+    ]
+    assert (non_gq_pop.groupby("address_id")["state", "puma"].nunique() == 1).values.all()
```
Comment (rmudambi) [inline integration_tests/test_household_structure.py:59]: Why do you need the `.values`?

**Review by rmudambi — APPROVED** [review]

File: integration_tests/test_household_structure.py (hunk 2 above)
Comment (stevebachmeier) [reply integration_tests/test_household_structure.py:59]: Just b/c we have two columns. I suppose I could have done `.all().all()` instead.

**Pull request merged**